### DISCUSSION TITLE
Lack of permission on sub-filesystems

### DISCUSSION THREAD

**Discussion by steel3d** [openzfsonwindows/openzfs#351] [General]
My zfs drive works fine if I just use the root file system, but if I create a sub-filesystem like 

>zfs create zfs1/Steve

Then Steve will not be accessible by certain means. 

Ex: if I'm logged in through remote desktop, with the same user I use locally, I can see the Steve folder in File Explorer, but I can not see its contents. Windows says "You don't currently have permission to access this folder" when I try to open it. 

Also, if I try to back up files to the Steve folder using SyncBackSE, I get errors on all nonexistent folders of the form: "Cannot create folder: K:\Steve\Test\ (The system cannot find the path specified.)"

I've tried granting full permissions to all users under the Security tab in the folder properties, to no avail. 

Is this a known issue, or am I missing something?

I wouldn't even bother using sub-filesystems, but I have not been able to figure out how to roll back to a snapshot on the root file system while keeping a backup of the latest state. With a sub-filesystem I can do it with cloning, promoting, and renaming. Is there a way to do this on the root filesystem? zfs rename doesn't seem to work on the root. 

By the way, renaming or destroying filesystems often causes my machine to hard reset. Not sure if this happens when the Steve folder is open in File Explorer or some other app. It doesn't always crash, so I can't file a good bug report. 

Appreciate any help!

**Comment by lundman** [comment]
I am in the middle of debugging this - you can set driverletter=on for your zfs1/Steve and you can use it.

**Comment by guenther-alka** [comment]
Have you set aclinherit=passthrough?

**Comment by steel3d** [comment]
I tried aclinherit. No go.

`zfs create -o aclinherit=passthrough zfs2/Steve`

driveletter works

`zfs create -o driveletter=M zfs2/Steve2`

But it doesn't fit too well with how I want to use the volume long term. I guess I can go ahead if you think I won't need to recreate the file system to fix it eventually @lundman .

**Comment by guenther-alka** [comment]
Aclinherit and drivelatter can be modified after creation. 
Driveletter needs a pool unmount/mount cycle.

**Comment by steel3d** [comment]
Hi @lundman , sorry for resurrecting this so late... Did you get any results from debugging this?

**Reply by lundman** [reply]
The mount/unmount was completely re-written at the start of 2025, and I'd like to believe it generally behaves. 
The permissions were indeed very wrong, worked on that in July. This one might not be completely fixed though, but at least the group is no longer completely wrong.

I don't know if the various combinations of mountpoint and driveletter has been ironed out. If you find a combination that doesn't work, it isn't hard for me to fix.

**Reply by sybreeder4** [reply]
Seems like this issue is still with latest version. 
i've tried with zfs set aclinherit=passthrough
And by putting all type of credentials but didn't worked when i want to use that pool as SMB and run app or play movie from that smb share from pool

**Comment by guenther-alka** [comment]
just to be sure:
set aclinherit does not modify acl of current files and folders.
It defines behaviour when you create a file or folder.

To modify acl of currenr files, use Windows Explorer > Properties > Security (as admin, you may need to take ownership)

**Comment by lundman** [comment]
So. It is not ACL, it is not sharename issues, it is not one of 5000 things I tried. The root of a sub-dataset do not want to work. But if you type in a lower directory, those work again.
